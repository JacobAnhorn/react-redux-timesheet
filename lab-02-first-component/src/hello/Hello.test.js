import React from 'react';
import { shallow } from 'enzyme';
import Hello from './Hello';
import renderer from 'react-test-renderer';

describe('Hello World', () => {
  it('Renders without exploding', () => {
    expect(shallow(<Hello />)).toHaveLength(1);
  });

  it('Should render with default text', () => {
    const component = shallow(<Hello />);

    expect(component).toIncludeText('Howdy');
    expect(component).toIncludeText('Partner');
  });

  it('Should render with our props', () => {
    const component = shallow(<Hello friend="Fred" />);

    expect(component).toIncludeText('Howdy');
    expect(component).toIncludeText('Fred');
    expect(component).not.toIncludeText('Partner');
  });
});

test('Link renders correctly', () => {
  const tree = renderer.create(<Hello friend="Bye" />).toJSON();
  expect(tree).toMatchSnapshot();
});
