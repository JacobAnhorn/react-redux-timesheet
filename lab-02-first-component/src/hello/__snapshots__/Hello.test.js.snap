// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="hello"
>
  <h1
    data-test-id="asdfasdf"
  >
    Howdy!!
  </h1>
  <h2>
    Bye
  </h2>
  <p>
    Congratulations! You have created your first React component!
  </p>
</div>
`;
